notebooks/examples/python/BritishNationalGrid.py

```diff
@@ -0,0 +1,370 @@
+# MAGIC You can think of Mosaic as a way to partition an overly complex row into multiple rows that have a balanced amount of computation each.
+
+# COMMAND ----------
+
+display(
+  postcodes_mosaic_frame.get_resolution_metrics(sample_rows=150)
+)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Indexing using the optimal resolution
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC We will use mosaic sql functions to index our points data. </br>
+# MAGIC Here we will use resolution -4 (500m), index resolution depends on the dataset in use. <br>
+# MAGIC There is a second best choice which is 4 (100m). <br> 
+# MAGIC The user can pass either numerical resolution or the string label to the index expressions. <br>
+# MAGIC BNG provides 2 types of hieracies. <br> 
+# MAGIC The standard hierarchy which operates with index resolutions in base 10 (i.e. (6, 1m), (5, 10m), (4, 100m), (3, 1km), (2, 10km), (1, 100km)) and index ids follow the format of letter pair followed by coordinate bins at the selected resolution (e.g. TQ100100 for (4, 100m)). <br>
+# MAGIC The quad hierachy (or quadrant hierarchy) which operates with index resolutions in base 5 (i.e. (-6, 5m), (-5, 50m), (-4, 500m), (-3, 5km), (-2, 50km), (-1, 500km)) and index ids follow the format of letter pair followed by coordinate bins at the selected resolution and folowed by quadrant letters (e.g. TQ100100SW for (-4, 500m)). Quadrants correspond to compas directions SW (south west), NW (north west), NE (north east) and SE (south east). <br>
+
+# COMMAND ----------
+
+uprns_table = spark.read.table("uprns_bng_table")
+uprns_table = (
+  uprns_table
+    .withColumn("uprn_bng_500m", mos.grid_pointascellid("uprn_point", lit(optimal_resolution)))
+    .withColumn("uprn_bng_500m_str", mos.grid_pointascellid("uprn_point", lit(optimal_resolution_str)))
+    .withColumn("uprn_bng_100m_str", mos.grid_pointascellid("uprn_point", lit("100m")))
+)
+
+# COMMAND ----------
+
+uprns_table.display()
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC Mosaic has a builtin wrappers for KeplerGL map plots using mosaic_kepler IPython magics. <br>
+# MAGIC Mosaic magics automatically handle bng idex system and CRS conversion for you. <br>
+# MAGIC Given that Kepler Plots are rendered on the browser side we are automatically limiting the row count to 1000. <br>
+# MAGIC The end user can override the number of ploted rows by specifying the desired number.
+
+# COMMAND ----------
+
+count_per_index = uprns_table.groupBy("uprn_bng_500m").count().cache()
+
+# COMMAND ----------
+
+# MAGIC %%mosaic_kepler
+# MAGIC count_per_index "uprn_bng_500m" "bng" 50
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC We will also index our postcodes using a built in generator function.
+
+# COMMAND ----------
+
+postcodes_with_index = (postcodes
+
+                           # We break down the original geometry in multiple smoller mosaic chips, each with its
```

Comment:
typo `smoller`
mosaic ->? tessellation (not sure about this)

Reply:
smoller is fixed in the last commit.
I would keep mosaic here.
We need to be a bit conscious about our naming standardisation and whether people would read well what we are saying. If we replace every mention of index with grid only you and I will understand this.
I will go through all of you suggestions and make some changes accordingly.
But some mention of index is needed. Grid is an index system, tessellation is just an indexing technique here.